### PR TITLE
feat: rename and document pipeline parameters

### PR DESCRIPTION
**Pull request by tumido**
Work in progress at the moment...

## Description
Resolves https://github.com/opendatahub-io/ilab-on-ocp/issues/125

1. Some params are renamed to align the pipeline with upstream InstructLab CLI.
2. Some params are cleaned up since they are not used at all.

Descriptions sourced mostly from:
- https://github.com/instructlab/instructlab/blob/fa5d1cc7df986b418112484c033a70b9a35a02c3/src/instructlab/configuration.py
- https://github.com/instructlab/eval/blob/f4db222e45f85add39f3b64571b27fb009ff4092/src/instructlab/eval/mt_bench.py#L177

## Known issues

RHOAI does not maintain param order defined in DSL, therefore in the UI params are ordered alphabetically. This means we'll get them in following order: `eval_*`, `mt_bench_*`, `sdg_*`, `train_*`. Not sure if this is what we want, but unless we rename them, there's nothing we can do. 

Result:

https://github.com/user-attachments/assets/527ec663-f08a-450c-8382-80beec524f6d

### CODE REVIEW

File: pipeline.py [hunk 1]
```diff
@@ -111,32 +110,65 @@ def pipeline_wrapper(mock: List[Literal[MOCKED_STAGES]]):
...
+        train_save_samples: int = SAVE_SAMPLES,
+        train_max_batch_len: int = MAX_BATCH_LEN,
+        train_seed: int = SEED,
+        # MT Bench
+        mt_bench_max_workers: str = MAX_WORKERS,
+        # Final evaluation
+        eval_few_shots: int = FEW_SHOTS,
+        eval_batch_size: str = BATCH_SIZE,
+        # Other options
+        merge_system_user_message: bool = MERGE_SYSTEM_USER_MESSAGE,
```
Comment (MichaelClifford) [inline pipeline.py:145]: This paramater is used for `mt_bench`
```suggestion
        mt_bench_merge_system_user_message: bool = MERGE_SYSTEM_USER_MESSAGE,
```

File: pipeline.py [hunk 2]
```diff
@@ -115,35 +114,71 @@ def pipeline_wrapper(mock: List[Literal[MOCKED_STAGES]]):
...
+        train_num_warmup_steps_phase_1: int = NUM_WARMUP_STEPS_PHASE_1,
+        train_num_warmup_steps_phase_2: int = NUM_WARMUP_STEPS_PHASE_2,
+        train_save_samples: int = SAVE_SAMPLES,
+        train_max_batch_len: int = MAX_BATCH_LEN,
+        train_seed: int = SEED,
+        # MT Bench
+        mt_bench_max_workers: str = MAX_WORKERS,
+        # Final evaluation
+        eval_few_shots: int = FEW_SHOTS,
+        eval_batch_size: str = BATCH_SIZE,
```
Comment (MichaelClifford) [inline pipeline.py:143]: should these both be `final_eval_` just to keep naming consistency with the step name?

File: pipeline.py (hunk 1 above)
Comment (tumido) [reply pipeline.py:145]: Yes, but it's used for MT Bench Branch in final eval as well:

https://github.com/opendatahub-io/ilab-on-ocp/blob/4ed3565cc053e4a65298c92c8c6556c8d3a7de67/eval/final/components.py#L371-L385

Should I split it into 2 separate parameters?

File: pipeline.py [hunk 3]
```diff
@@ -111,32 +110,65 @@ def pipeline_wrapper(mock: List[Literal[MOCKED_STAGES]]):
...
+        # Other options
+        merge_system_user_message: bool = MERGE_SYSTEM_USER_MESSAGE,
+        storage_class_name: str = "nfs-csi",
```
[the remark this answers is not in the document]
Comment (tumido) [reply pipeline.py:146]: 🤷 I would vote for `k8s_` prefix if any.. but I really don't think we need to make the names any longer...

File: pipeline.py (hunk 1 above)
Comment (MichaelClifford) [reply pipeline.py:145]: yeah. I think we should separate it. We should not make the assumption that 2 similar but different process always need the same params.

File: pipeline.py (hunk 2 above)
Comment (tumido) [reply pipeline.py:143]: I have no preference... does it convey any message to the end user? After all, they will get this context from the description field anyways...
image

File: pipeline.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MichaelClifford) [reply pipeline.py:146]: My only worry is that they will stand out as not part of any category and will just appear in alphabetical order. But if its the only one, then maybe its fine and its category is just `storage_`?

**Review by MichaelClifford — APPROVED** [review]

**Pull request merged**